src/Input/Input.component.tsx

```diff
@@ -185,19 +186,24 @@ export const Input: React.FunctionComponent<Props> = ({
     <ThemeProvider theme={(outerTheme: any) => outerTheme || theme}>
       <>
         <SInputWrapper
-          className={
-            (inputProps.icon || inputProps.addonText) &&
-            `input-icon-${inputProps.iconPosition ||
-              inputProps.addonTextPosition}`
-          }
+          className={classNames(
+            [
+              (inputProps.icon || inputProps.addonText) &&
+                `input-icon-${inputProps.iconPosition ||
+                  inputProps.addonTextPosition}`,
+            ],
+            {
+              disabled: inputProps.disabled,
+            },
+          )}
           {...inputProps}
         >
           {('left' === inputProps.iconPosition ||
             'left' === inputProps.addonTextPosition) && (
             <span>{inputProps.icon || inputProps.addonText}</span>
           )}
           <SInput
-            value={value}
+            value={(value || '').substring(0, inputProps.maxLength)}
```

Comment:
The maxLength does not seem to work either

![image](https://user-images.githubusercontent.com/20873103/58332460-f03dc780-7e43-11e9-9a2b-78b5b459aefc.png)

Reply:
Works as expected